sno/working_copy.py

```diff
@@ -90,6 +91,26 @@ def __str__(self):
             return f"Working Copy is tree {self.working_copy_tree_id}; expecting {self.match_tree_id}"
 
 
+class SQLCommand(Enum):
+    INSERT = "INSERT"
+    INSERT_OR_REPLACE = "INSERT OR REPLACE"
+
+
+def sql_insert_dict(dbcur, sql_command, table_name, row_dict):
+    """
+    Inserts a row into a database table.
+    sql_command should be a member of SQLCommand (INSERT or INSERT_OR_REPLACE)
+    """
+    keys, values = zip(*row_dict.items())
+    sql = f"""
+        {sql_command.value} INTO {table_name}
+            ({','.join([gpkg.ident(k) for k in keys])})
+        VALUES
+            ({','.join(['?'] * len(keys))});
+    """
+    dbcur.execute(sql, values)
```

Comment:
Can we make this return the cursor so it acts like [`Cursor.execute()`](https://rogerbinns.github.io/apsw/cursor.html?highlight=insert%20return#apsw.Cursor.execute) itself?

Reply:
Sure, done.